packages/react/src/components/Modal/Modal.mdx

```diff
@@ -1,13 +1,295 @@
-import { Props } from '@storybook/addon-docs/blocks';
+import { Story, Props } from '@storybook/addon-docs/blocks';
+import Modal from '../Modal';
 
 # Modal
 
 ## Overview
 
+`<Modal>` allows you to use its bespoke set of contents. `children` prop maps to
+the modal body content. You can also use `modalLabel`, `modalHeading`,
+`secondaryButtonText` and `primaryButtonText` props to change the corresponding
+text.
+
+<Preview>
+  <Story id="modal--with-state-manager" />
+</Preview>
+
 ## Component API
 
 <Props />
 
+## Opening/closing modal
+
+For both modal variants, you can open/close the modal by changing the `open`
+prop. For example, you can implement a launcher button with a React state and a
+`<Button>` that changes the state:
+
+```javascript
+function ModalStateManager() {
+  const [open, setOpen] = useState(false);
+  return (
+    <>
+      {typeof document === 'undefined'
+        ? null
+        : ReactDOM.createPortal(
+            <ComposedModal open={open} onClose={() => setOpen(false)}>
+              ...
+            </ComposedModal>,
+            document.body
+          )}
+      <Button kind="primary" onClick={() => setOpen(true)}>
+        Open modal
+      </Button>
+    </>
+  );
+}
+```
+
+You can create an abstract version of such state manager, as shown below.
+
+```javascript
+const ModalStateManager = ({
+  renderLauncher: LauncherContent,
+  children: ModalContent,
+}) => {
+  const [open, setOpen] = useState(false);
+  return (
+    <>
+      {!ModalContent || typeof document === 'undefined'
+        ? null
+        : ReactDOM.createPortal(
+            <ModalContent open={open} setOpen={setOpen} />,
+            document.body
+          )}
+      {LauncherContent && <LauncherContent open={open} setOpen={setOpen} />}
+    </>
+  );
+};
+```
+
+`<ModalWrapper>` is a modal state manager implementation which ships out of the
+box with Carbon.
+
+<InlineNotification kind="warning">
```

Comment:
We need to either add
`import { InlineNotification } from '../Notification';` to the top of the `mdx` file or we can just use a quite block to reinforce emphasis. Notification is not rendering as-is for me.

```mdx
> `<Modal>` / `<ComposedModal>` has to be put at the top level in a React tree.
> The easiest way to ensure that is using React portal as shown in above
> example.
```

Reply:
added import to the mdx file